trace_decoder/tests/trace_decoder_tests.rs

```diff
@@ -0,0 +1,266 @@
+//! Tests to check the parsing/decoding and `GenerationInputs` validity.
+//! They rely on the jerigon and cdk erigon witness files as input.
+
+use std::time::Duration;
+use std::{
+    fs,
+    path::{Path, PathBuf},
+};
+
+use alloy_rpc_types_eth::Header;
+use anyhow::Context as _;
+use evm_arithmetization::prover::testing::simulate_execution;
+use evm_arithmetization::GenerationInputs;
+use itertools::Itertools;
+use log::info;
+use mpt_trie::partial_trie::PartialTrie;
+use plonky2::field::goldilocks_field::GoldilocksField;
+use plonky2::util::timing::TimingTree;
+use plonky2_maybe_rayon::*;
+use pretty_env_logger::env_logger::{try_init_from_env, Env, DEFAULT_FILTER_ENV};
+use prover::BlockProverInput;
+use rstest::rstest;
+use trace_decoder::OtherBlockData;
+
+type F = GoldilocksField;
+
+const JERIGON_WITNESS_DIR: &str = "tests/data/witnesses/zero_jerigon";
+///TODO Add CDK Erigon witness test data.
+/// Local [cdk erigon](https://github.com/0xPolygonHermez/cdk-erigon?tab=readme-ov-file#running-cdk-erigon) dev network
+/// could be used for basic witness generation.
+/// Related work for type2 prover is on the [type2_cancun](https://github.com/0xPolygonZero/zk_evm/pull/319) branch at the moment.
+/// When the cdk erigon witness data is added, enable test execution for
+/// `CDK_ERIGON_WITNESS_DIR`
+//const CDK_ERIGON_WITNESS_DIR: &str =
+// "tests/data/witnesses/hermez_cdk_erigon";
+
+enum JsonFileType {
+    Witness,
```

Comment:
Nonblocking: why does this exist if there's only one variant?

Reply:
There were two variants (Witness and Header). I thought searching for headers might still be needed. But you are right, I'll remove it until proven otherwise.